compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/interop/LocalUIViewController.uikit.kt

```diff
@@ -17,13 +17,36 @@
 package androidx.compose.ui.interop
 
 import androidx.compose.runtime.staticCompositionLocalOf
+import kotlin.native.ref.WeakReference
 import platform.UIKit.UIViewController
 
 /**
  * public value to get UIViewController of Compose window for library authors.
  * Maybe useful for features, like VideoPlayer and Bottom menus.
  * Please use it careful and don't remove another views.
+ *
+ * _IMPORTANT NOTE_:
+ *
+ * Never capture the unwrapped strong reference in any closure inside Composition. Doing so will cause
+ * a memory leak.
+ *
+ * Unwrap it only in the place of usage, for example:
+ * ```
+ * @Composable
+ * private fun Foo() {
+ *     // can't use LocalUIViewController.current inside onClick because onClick is not @Composable
+ *     val viewController = LocalUIViewController.current
+ *
+ *     Column {
+ *         Button(onClick = {
+ *             viewController.get()?.presentViewController(...)
+ *         }) {
+ *             Text("Push")
+ *         }
+ *     }
+ * }
+ * ```
  */
-val LocalUIViewController = staticCompositionLocalOf<UIViewController> {
+val LocalUIViewController = staticCompositionLocalOf<WeakReference<UIViewController>> {
```

Comment:
It will break backward compatibility, but it is fine, because we are in alpha status.

For example library "Moko" may use this value. @Alex009 please take a look.

Reply:
Yes, it's ok on this stage, but please explicitly describe it in PR description (it will be visible in our changelog) in `API Change` section

Reply:
Yes this change is ok

Reply:
Added `API change` section.

Reply:
@Alex009, we decided to revert it back to the strong reference, and fix issue another way [here](https://github.com/JetBrains/compose-multiplatform-core/pull/747)